packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -62,6 +63,140 @@ const TextStyle _kToolbarButtonDisabledFontStyle = TextStyle(
 // Eyeballed value.
 const EdgeInsets _kToolbarButtonPadding = EdgeInsets.symmetric(vertical: 10.0, horizontal: 18.0);
 
+// Generates the child that's passed into CupertinoTextSelectionToolbar.
+class _CupertinoTextSelectionToolbarWrapper extends StatefulWidget {
+  const _CupertinoTextSelectionToolbarWrapper({
+    Key key,
+    this.barTopY,
+    this.arrowTipX,
+    this.handleCut,
+    this.handleCopy,
+    this.handlePaste,
+    this.handleSelectAll,
+    this.isArrowPointingDown,
+  }) : super(key: key);
+
+  final double arrowTipX;
+  final double barTopY;
+  final VoidCallback handleCut;
+  final VoidCallback handleCopy;
+  final VoidCallback handlePaste;
+  final VoidCallback handleSelectAll;
+  final bool isArrowPointingDown;
+
+  @override
+  _CupertinoTextSelectionToolbarWrapperState createState() => _CupertinoTextSelectionToolbarWrapperState();
+}
+
+class _CupertinoTextSelectionToolbarWrapperState extends State<_CupertinoTextSelectionToolbarWrapper> with WidgetsBindingObserver {
+  // Whether the user's system clipboard is empty as of the last check.
+  bool _isClipboardEmpty;
+
+  // Asynchronously check if theh clipboard has data and update
+  // _isClipboardEmpty if needed.
+  void _updateClipboardStatus() {
+    Clipboard.getData(Clipboard.kTextPlain).then((ClipboardData value) {
+      setState(() {
+        if (!mounted) {
+          return;
+        }
+        final bool isClipboardEmpty = value?.text == null || value.text.isEmpty;
+        if (isClipboardEmpty == _isClipboardEmpty) {
+          return;
+        }
+        _isClipboardEmpty = isClipboardEmpty;
+      });
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+    // Asynchronously check if the clipboard has changed, and trigger another
+    // build if so.
+    _updateClipboardStatus();
```

Comment:
should we only do this if `widget.handlePaste` is set to something? (here and elsewhere where we call it)

Reply:
(plus we'd need to do it in didUpdateWidget if it changes)

Reply:
Good call, I'm going to change this to check if widget.handlePaste is set and do this in didUpdateWidget as well. You can actually cause a bug without this change by opening the menu with nothing copied, copying something while the menu is open, and then tapping Select All to make the menu rebuild.  At least in the simulator it's possible.